rasa/core/tracker_store.py

```diff
@@ -161,18 +162,18 @@ def create(
         import pymongo.errors
         import sqlalchemy.exc
 
+        if isinstance(obj, EndpointConfig):
+            result = plugin_manager().hook.update_endpoint_config(endpoint_config=obj)
```

Comment:
should we move this hook into `Endpoint.__init__`? this would allow other kinds of endpoints to also use this hook.

Reply:
Problem is that update of the endpoint depends on what endpoint represents.
Tracker store's endpoint has different set of attributes from event broker's endpoint.
Thus we use different approach when we update endpoint config for tracker store from how we will update endpoint for event broker.
I updated the signature of update_endpoint_config hook to:
```python
update_endpoint_config(endpoint_config: EndpointConfig, endpoint_type: Text)
```

Call to it would be:
```python
update_endpoint_config(endpoint_config=endpoint_config, endpoint_type="tracker_store")
update_endpoint_config(endpoint_config=endpoint_config, endpoint_type="event_broker")
```

Reply:
@tmbo 
Scratch my last comment.
I was able to remove entire `update_endpoint_config` hook.
Now everything is seamlessly handled by `get_auth_retry_wrapper` hook and I had to change it's signature to:
```python
@hookspec(firstresult=True)  # type: ignore[misc]
def get_auth_retry_wrapper(  # type: ignore[empty-body]
    endpoint_config: Union["TrackerStore", "EndpointConfig"],
    domain: "Domain",
    event_broker: Optional["EventBroker"] = None,
) -> "TrackerStore":
```

Reply:
We cannot move this to `EndpointConfig.__init__` at this stage because we need original Endpoint configuration when we need to re-fetch information for certain fields. If we would move this to `EndpointConfig.__init__` we would loose this information.